libpod/container_inspect.go

```diff
@@ -196,7 +197,11 @@ func (c *Container) getContainerInspectData(size bool, driverData *define.Driver
 			// An error here is not considered fatal; no health state will be displayed
 			logrus.Error(err)
 		} else {
-			data.State.Health = healthCheckState
+			if reflect.DeepEqual(healthCheckState, define.HealthCheckResults{}) {
```

Comment:
why are we using DeepEqual here? first it slow but second we already are checking if there is healthcheck with
`c.config.HealthCheckConfig != nil` so we should just assign the struct here.
The health key should only be missing if there is no healthcheck but not when the log is just empty (like it is doing here)